### PR TITLE
refactor; add tests

### PR DESCRIPTION
**Pull request by rogpeppe**
Still very much a work in progress with quite a few features
missing, but should be useful nonetheless.

Main differences from http(1) - no output highlighting,
output remains the same regardless of whether stdout
is directed to a tty or not. Does not print headers by default;
-h flag prints headers in addition to body; -B flag suppresses
body output.

### CODE REVIEW

File: http.go [hunk 1]
```diff
@@ -12,31 +14,18 @@ import (
...
-	body       bool
-	rjson      bool
-	raw        bool
-	noBrowser  bool
-	basicAuth  string
-	cookieFile string
-
-	// TODO auth, verify, proxy, file, timeout
-}
-
```
Comment (mhilton) [inline http.go:39]: The helpMessage (below) is out of date and could do with sorting out.

File: http.go [hunk 2]
```diff
@@ -94,134 +83,254 @@ const helpMessage = `usage: http [flag...] [METHOD] URL [REQUEST_ITEM [REQUEST_I
...
 
-	flag.StringVar(&flags.basicAuth, "a", "", "http basic auth (username:password)")
-	flag.StringVar(&flags.basicAuth, "auth", "", "")
+	fset.BoolVar(&p.raw, "raw", false, "print response body without any JSON post-processing")
 
-	flag.StringVar(&flags.cookieFile, "cookiefile", filepath.Join(os.Getenv("HOME"), ".go-cookies"), "file to store persistent cookies in")
+	fset.StringVar(&p.basicAuth, "a", "", "http basic auth (username:password)")
+	fset.StringVar(&p.basicAuth, "auth", "", "")
+
+	fset.StringVar(&p.cookieFile, "cookiefile", filepath.Join(os.Getenv("HOME"), ".go-cookies"), "file to store persistent cookies in")
```
Comment (mhilton) [inline http.go:205]: There are some fairly good reasons not to add even more stuff into $HOME $XDG_DATA_HOME might be more appropriate.

**Comment by mhilton** [comment]
:+1: But just a note that we have some use cases where we need to get and discharge a macaroon before doing the HTTP request as not having a macaroon is not a failure, just a different response path.

File: http.go (hunk 2 above)
Comment (rogpeppe) [inline http.go:205]: If XDG_DATA_HOME was not platform-specific, I'd be tempted to agree. but it's not even set for me under ubuntu linux.

One possibility might be to require that a GO_COOKIEFILE environment variable be set; if it's not, we'll print a warning and use in-memory cookies.

File: http.go (hunk 1 above)
Comment (rogpeppe) [inline http.go:39]: Not really, but I'll probably change the examples in a future PR, when we've got a better name for this command.

**Pull request merged**